rust/src/lib/nm/nm_dbus/gen_conf/route_rule.rs

```diff
@@ -23,10 +25,26 @@ impl NmIpRouteRule {
                 keys.push(to_str);
             }
 
-            let from_str = match (self.from.as_ref(), self.from_len.as_ref()) {
-                (Some(f), Some(f_len)) => format!("from {f}/{f_len}"),
-                (Some(f), None) => format!("from {f}"),
-                _ => "".to_string(),
+            let mut from_str =
+                match (self.from.as_ref(), self.from_len.as_ref()) {
+                    (Some(f), Some(f_len)) => format!("from {f}/{f_len}"),
+                    (Some(f), None) => format!("from {f}"),
+                    _ => "".to_string(),
+                };
+            from_str = match (self.from.as_ref(), self.to.as_ref()) {
+                (None, None) => {
+                    if let Some(family) = self.family {
+                        if family == AF_INET {
+                            "from 0.0.0.0/0".to_string()
+                        } else {
+                            "from ::/0".to_string()
+                        }
+                    } else {
```

Comment:
How about do this in `Deserialize` of `RouteRuleEntry` or `pre_edit_cleanup()`? This is not NM specific.

Reply:
This is NM keyfile specific. In kernel this is not needed at all but nmcli adds it to the keyfile this is why we are doing this only for keyfile. We do not want to add the default ip always because it is not necessary and I would prefer to follow iproute2 format which is a standard somehow, that is:

```yaml
route-rules:
  config:
    - route-table: 254
      priority: 100
      family: ipv4
```

We do not want `ip-from: 0.0.0.0/0` there. I will ask NM people while are they adding the default ip address to keyfile if it is not needed at all.